packages/slider/index.tsx

```diff
@@ -0,0 +1,677 @@
+    if (valueNeedsToBeFlipped) {
+      delta = -delta;
+    }
+
+    switch (keyId) {
+      case KEY_IDS.ARROW_LEFT:
+      case KEY_IDS.ARROW_DOWN:
+        return this.foundation.getValue() - delta;
+      case KEY_IDS.ARROW_RIGHT:
+      case KEY_IDS.ARROW_UP:
+        return this.foundation.getValue() + delta;
+      case KEY_IDS.HOME:
+        return min;
+      case KEY_IDS.END:
+        return max;
+      case KEY_IDS.PAGE_UP:
+        return this.foundation.getValue() + delta * numbers.PAGE_FACTOR;
+      case KEY_IDS.PAGE_DOWN:
+        return this.foundation.getValue() - delta * numbers.PAGE_FACTOR;
+      default:
+        return NaN;
+    }
+  }
+
+  onFocus = () => {
+    if (this.preventFocusState) {
+      return;
+    }
+    this.adapter.addClass(cssClasses.FOCUS);
+  };
+
+  onBlur = () => {
+    this.preventFocusState = false;
+    this.adapter.removeClass(cssClasses.FOCUS);
+  };
+
+  onThumbDown = () => (this.handlingThumbTargetEvt = true);
+
+  render() {
+    const {
+      discrete,
+      displayMarkers,
+      value,
+      min,
+      max,
+      step,
+      disabled,
+      /* eslint-disable @typescript-eslint/no-unused-vars */
+      className,
+      isRtl,
+      onValueChange,
+      onValueInput,
+      /* eslint-enable @typescript-eslint/no-unused-vars */
+      ...otherProps
+    } = this.props;
+    const {markerValue, markerStyles} = this.state;
+
+    return (
+      <div
+        className={this.classes}
+        role='slider'
+        aria-valuemin={min}
+        aria-valuemax={max}
+        aria-valuenow={value}
+        data-step={step}
+        aria-label='Select Value'
+        aria-disabled={disabled}
+        ref={this.sliderElement}
+        onBlur={this.onBlur}
+        onFocus={this.onFocus}
+        onMouseDown={this.onDown}
+        onPointerDown={this.onDown}
+        onTouchStart={this.onDown}
+        onKeyDown={this.onKeyDown}
+        {...otherProps as React.HTMLProps<HTMLDivElement>}
+      >
+        <div className='mdc-slider__track-container'>
+          <div className='mdc-slider__track' ref={this.trackElement} />
+          {discrete && displayMarkers && (
+            <div className='mdc-slider__track-marker-container'>
+              {markerStyles.map((styles, index) => (
+                <div
+                  key={index}
+                  className='mdc-slider__track-marker'
+                  style={styles}
+                />
+              ))}
+            </div>
+          )}
+        </div>
+        <div
+          className='mdc-slider__thumb-container'
+          ref={this.thumbContainerElement}
+          onMouseDown={this.onThumbDown}
+          onPointerDown={this.onThumbDown}
+          onTouchStart={this.onThumbDown}
+          onTransitionEnd={this.onThumbEnd}
+        >
+          {discrete && (
+            <div className='mdc-slider__pin'>
+              <span className='mdc-slider__pin-value-marker'>
+                {markerValue}
```

Comment:
The markerValue in the demo is a float/has many decimal places. I believe this is a bug. Please address

Reply:
I don't see this when rendering in Chrome, are you saying it has decimal places in the screenshot?

Reply:
Yes, as you move the slider it displays decimal places.

Reply:
This is what I see in Chrome on macOS:
![image](https://user-images.githubusercontent.com/6323077/59272627-67d16c00-8c24-11e9-95f5-7f0628cdd941.png)

If it's still happening after I removed the `setValue` function in https://github.com/material-components/material-components-web-react/pull/868/commits/92d43bec99de0cdcda193cca2fefb78a3722d86d, then I can try to add code to truncate the value displayed by React.

I guess we'll have to account for the fact that the step itself can be a floating point number.